### PR TITLE
Allow ELResolver to be specified in Config

### PR DESCRIPTION
**Pull request by mattcoley**
Alternative to https://github.com/HubSpot/jinjava/pull/431

This will allow us to connect our custom ELResolver.

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/JinjavaConfig.java [hunk 1]
```diff
@@ -307,7 +309,10 @@ public Builder withMaxMacroRecursionDepth(int maxMacroRecursionDepth) {
     }
 
     public Builder withReadOnlyResolver(boolean readOnlyResolver) {
```
Comment (mattcoley) [inline src/main/java/com/hubspot/jinjava/JinjavaConfig.java:311]: I wanted to keep the existing behavior consistent so kept this method in for the builder.

**Comment by boulter** [comment]
instead of making it part of the config, how about allowing it to be set on the Jinjava instance like it is for `ResourceLocator`. Then you can create whatever implementation you want.

**Comment by mattcoley** [comment]
The ELResolver used right now is determined by the config (through the `readOnlyResolver` setting). Is there any advantage to exposing the ELResolver at the Jinjava level? That would mean you would need a separate instance of Jinjava for every ELResolver you would like to use vs. being able to reuse the same Jinjava object with different ELResolver's specified by the config.

**Comment by mattcoley** [comment]
I realized I forgot to add the Builder method to the config for specifying your own ELResolver.

**Comment by boulter** [comment]
I think it should be possible to provide your own chain of resolvers. The readOnlyResolver seems a little too inflexible. Maybe you could just pass your `CompositeELResolver` to config if you don't want to add a setter to `Jinjava`.

**Comment by mattcoley** [comment]
The config accepts any `ELResolver` so you can pass in your own `CompositeELResolver` to the config. The `readOnlyResolver` boolean was an existing config value that I didn't want to remove for backwards compatibility, I just moved the logic to map the boolean value to one of the two default values to the config from the JinjavaInterpreterResolver

**Pull request merged**